server/src/main/java/org/elasticsearch/common/blobstore/BlobPath.java

```diff
@@ -82,4 +97,21 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(paths);
     }
+
+    public enum Purpose {
+        GENERIC("Generic"),
```

Comment:
Could we call this `SNAPSHOT` instead of `GENERIC`? That way if we have another purpose in future we will have to add another entry here, because none of the existing entries will fit.